sklearn/utils/estimator_checks.py

```diff
@@ -878,15 +878,27 @@ def check_array_api_input(
     X_xp = xp.asarray(X, device=device)
     y_xp = xp.asarray(y, device=device)
 
-    est.fit(X, y)
+    X_types = est._get_tags().get("X_types", [""])
+    if "labels" in X_types[0]:
```

Comment:
Cannot it happen that `X_types` has more than one element and the labels type is not in the first position in the list?

Reply:
If that is the case then how can I handle this scenario? @ogrisel @betatim